opimodel/widgets.py

```diff
@@ -254,30 +258,31 @@ def __init__(self, x, y, width, height, text):
         self.label = text
 
 
-class ToggleButton(Widget):
+class ToggleButton(ActionWidget):
 
     TYPE_ID = 'org.csstudio.opibuilder.widgets.BoolButton'
 
     def __init__(self, x, y, width, height, on_text, off_text):
         super(ToggleButton, self).__init__(
             ToggleButton.TYPE_ID, x, y, width, height)
 
-        self.actions = []
         self.on_label = on_text
         self.off_label = off_text
         self.toggle_button = True
         self.effect_3d = True
         self.square_button = True
         self.show_boolean_label = True
         self.show_led = False
+        self.push_action_index = 0
+        self.released_action_index = 1
 
     def add_push_action(self, action):
-        self.actions.append(action)
+        self.actions.add_action(action)
         self.push_action_index = len(self.actions) - 1
```

Comment:
OK, I was confused.

Looking deeper at a BoolButton added in the OPIEditor it has:
    `<actions hook="false" hook_all="false">`
As ToggleButton extends ActionWidget it picks up the default args of `hook_first=True, hook_all=False` and doesn't set `hook` at all.